daemons/attrd/attrd_commands.c

```diff
@@ -886,6 +886,12 @@ attrd_peer_update(crm_node_t *peer, xmlNode *xml, const char *host, bool filter)
         v->current = (value? strdup(value) : NULL);
         a->changed = TRUE;
 
+        if (pcmk__str_eq(host, attrd_cluster->uname, pcmk__str_casei)
+            && pcmk__str_eq(attr, XML_CIB_ATTR_SHUTDOWN, pcmk__str_none)
+            && !pcmk__str_eq(value, "0", pcmk__str_null_matches)) {
+            attrd_set_requesting_shutdown();
```

Comment:
I'm thinking to stay on the safe side, we should handle the value flipping from nonzero back to zero before the DC has a chance to schedule shutdown (i.e. clear requesting shutdown if the value is 0).

Reply:
Makes sense.